pkg/storagev2/storageutil/storageutil.go

```diff
@@ -0,0 +1,18 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package storageutil
+
+import (
+	"strings"
+)
+
+// JoinFields joins the given fields with a slash.
+// If any of the fields contains a slash, it is escaped.
+func JoinFields(fields ...string) string {
+	for i, field := range fields {
+		fields[i] = strings.ReplaceAll(field, "/", "%2F")
```

Comment:
If the motivation behind adding '/' between fields is to make it readable/printable, this is not possible because we convert everything to bytes anyway, and it would require a step of decoding the ID to make it printable.

Reply:
> The original issue is that if the field's byte slice contains a single byte with the 47 value (which is the '/' ascii character, the path.join would treat it as part of the path and escape it.

This is not true, `path.Join` will remove the `/` character as it calls `path.Clean` which replaces multiple slashes with a single slash.

> Why can't we just use the good ol' string + string concatenation.

We can, but it's inefficient and doesn't provide a clear visual separator when debugging.

Removed the escape part.